build.go

```diff
@@ -139,21 +143,52 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return err
 	}
 
-	return c.lifecycle.Execute(ctx, build.LifecycleOptions{
+	lifecycleOpts := build.LifecycleOptions{
 		AppPath:            appPath,
 		Image:              imageRef,
 		Builder:            ephemeralBuilder,
 		RunImage:           runImageName,
 		ClearCache:         opts.ClearCache,
 		Publish:            opts.Publish,
+		UseCreator:         false,
+		LifecycleImage:     ephemeralBuilder.Name(),
 		HTTPProxy:          proxyConfig.HTTPProxy,
 		HTTPSProxy:         proxyConfig.HTTPSProxy,
 		NoProxy:            proxyConfig.NoProxy,
 		Network:            opts.ContainerConfig.Network,
 		Volumes:            platformVolumes,
 		DefaultProcessType: opts.DefaultProcessType,
 		FileFilter:         opts.FileFilter,
-	})
+	}
+
+	// Technically the creator is supported as of platform API version 0.3 (lifecycle version 0.7.0+) but earlier versions
+	// have bugs that make using the creator problematic.
+	lifecycleSupportsCreator := !ephemeralBuilder.LifecycleDescriptor().Info.Version.LessThan(semver.MustParse("0.7.4"))
+
+	if lifecycleSupportsCreator && (!opts.Publish || opts.TrustBuilder) {
+		// no need to fetch a lifecycle image, it won't be used
+		lifecycleOpts.UseCreator = true
+		return c.lifecycle.Execute(ctx, lifecycleOpts)
+	}
+
+	lifecycleImageSupported := !ephemeralBuilder.LifecycleDescriptor().Info.Version.LessThan(semver.MustParse("0.7.5"))
```

Comment:
I think it should be 0.7.4.

Reply:
At this time we will start creating lifecycle images starting at 0.7.5. I wasn't part of the discussion but I'm not sure it makes sense to provide older lifecycle images. A quick look at the current suggested builders the all use 0.7.5 with the exception of the heroku builder (version 0.6.2).

Reply:
I don't think we intend to publish a `0.7.4` lifecycle image - that version of the lifecycle is broken when you use the 5 phases with `--publish` (under certain conditions) and that is the exact scenario when we want to use the lifecycle image.

If someone provides lifecycle 0.7.4 and they trust their builder or they are not publishing, then great - we'll use the creator. But if not then we'll warn them (since lifecycle image is not available) and we'll use that lifecycle with the 5 phases. Their build might fail due to the bug, but sadly there isn't anything we can do about that.